src/main/java/teammates/ui/controller/InstructorEditInstructorFeedbackSaveAction.java

```diff
@@ -0,0 +1,195 @@
+ * save the data after moderating the instructor.
+ */
+public class InstructorEditInstructorFeedbackSaveAction extends FeedbackSubmissionEditSaveAction {
+    InstructorAttributes moderatedInstructor;
+    
+    /**
+     * Verifies if the user is allowed to carry out the action.
+     */
+    @Override
+    protected void verifyAccesibleForSpecificUser() {
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, account.googleId);
+        FeedbackSessionAttributes session = logic.getFeedbackSession(feedbackSessionName, courseId);
+
+        new GateKeeper().verifyAccessible(instructor,
+                session,
+                false, 
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS);
+    }
+    
+    /**
+     * Retrieves any additional parameters from request and set them accordingly.
+     */
+    @Override
+    protected void setAdditionalParameters() {
+        String moderatedInstructorEmail = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_MODERATED_INSTRUCTOR);
+        Assumption.assertPostParamNotNull(Const.ParamsNames.FEEDBACK_SESSION_MODERATED_INSTRUCTOR, moderatedInstructorEmail);
+
+        moderatedInstructor = logic.getInstructorForEmail(courseId, moderatedInstructorEmail);
+    }
+    
+    /**
+     * Checks if the instructor only submitted responses that he/she should be submitting when moderating.
+     */
+    @Override
+    protected void checkAdditionalConstraints() {
+        // check the instructor did not submit responses to questions that he/she should not be able when moderating
+        
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, account.googleId);
+        int numOfQuestionsToGet = data.bundle.questionResponseBundle.size();
+
+        for (int questionIndx = 1; questionIndx <= numOfQuestionsToGet; questionIndx++) {
+            String paramMapKey = Const.ParamsNames.FEEDBACK_QUESTION_ID + "-" + questionIndx;
+            String questionId = HttpRequestHelper.getValueFromParamMap(requestParameters, paramMapKey);
+            
+            if (questionId == null) {
+                // we do not throw an error if the question was not present on the page for instructors to edit
+                continue;
+            }
+            
+            FeedbackQuestionAttributes questionAttributes = data.bundle.getQuestionAttributes(questionId);
+            
+            if (questionAttributes == null) {
+                statusToUser.add(new StatusMessage("The feedback session or questions may have changed while you were submitting. "
+                                                + "Please check your responses to make sure they are saved correctly.", StatusMessageColor.WARNING));
+                isError = true;
+                log.warning("Question not found. (deleted or invalid id passed?) id: "+ questionId + " index: " + questionIndx);
```

Comment:
This warning message lacks details to be useful. Should also include the course and feedback session.